src/pages/o/[orgId]/campaigns.tsx

```diff
@@ -1,55 +1,70 @@
 import { GetServerSideProps } from 'next';
+import { dehydrate } from 'react-query/hydration';
+import { QueryClient, useQuery } from 'react-query';
 
-export const getServerSideProps : GetServerSideProps = async (context) => {
-    let props;
-
-    try {
-        const { orgId } = context.params;
-
-        const cRes = await fetch(`http://api.zetk.in/v1/orgs/${orgId}/campaigns`);
-        const cData = await cRes.json();
-        const oRes = await fetch(`https://api.zetk.in/v1/orgs/${orgId}`);
-        const oData = await oRes.json();
-
-        props = {
-            campaigns: cData.data,
-            org: oData.data,
-        };
-    }
-    catch (err) {
-        if (err.name != 'FetchError') {
-            throw err;
+function getCampaigns(orgId) {
+    return async () => {
+        try {
+            const cRes = await fetch(`http://localhost:3000/api/orgs/${orgId}/campaigns`);
+            const cData = await cRes.json();
+
+            return cData.data;
+        }
+        catch (err) {
+            if (err.name != 'FetchError') {
+                throw err;
+            }
+            return null;
+        }
+    };
+}
+
+function getOrg(orgId) {
+    return async () => {
+        try {
+            const oRes = await fetch(`http://localhost:3000/api/orgs/${orgId}`);
+            const oData = await oRes.json();
+
+            return oData.data;
         }
-    }
-
-    if (props) {
-        return { props };
-    }
-    else {
-        return {
-            notFound: true,
-        };
-    }
+        catch (err) {
+            if (err.name != 'FetchError') {
+                throw err;
+            }
+            return null;
+        }
+    };
+}
+
+export const getServerSideProps : GetServerSideProps = async (context) => {
+    const queryClient = new QueryClient();
+    const { orgId } = context.params;
+
+    await queryClient.prefetchQuery('campaigns', getCampaigns(orgId));
```

Comment:
This is storing the list of campaigns in the cache without any information about what organization they belong to. Browsing between two different organizations will reuse campaign information from the cache, even though the content should be different.